src/agoradatatools/process.py

```diff
@@ -16,59 +16,59 @@
-        return transform.transform_distribution_data(
-            datasets=datasets,
-            overall_max_score=dataset_obj["custom_transformations"][
-                "overall_max_score"
-            ],
-            genetics_max_score=dataset_obj["custom_transformations"][
-                "genetics_max_score"
-            ],
-            omics_max_score=dataset_obj["custom_transformations"]["omics_max_score"],
-        )
-    if dataset_name == "team_info":
-        return transform.transform_team_info(datasets=datasets)
-    if dataset_name == "rnaseq_differential_expression":
-        return transform.transform_rnaseq_differential_expression(datasets=datasets)
-    if dataset_name == "gene_info":
-        return transform.transform_gene_info(
-            datasets=datasets,
-            adjusted_p_value_threshold=dataset_obj["custom_transformations"][
-                "adjusted_p_value_threshold"
-            ],
-            protein_level_threshold=dataset_obj["custom_transformations"][
-                "protein_level_threshold"
-            ],
-        )
-    if dataset_name == "rna_distribution_data":
-        return transform.transform_rna_distribution_data(datasets=datasets)
-    if dataset_name == "proteomics_distribution_data":
-        return transform.transform_proteomics_distribution_data(datasets=datasets)
-    if dataset_name in ["proteomics", "proteomics_tmt", "proteomics_srm"]:
-        df = datasets[dataset_name]
-        return transform.transform_proteomics(df=df)
-    if dataset_name in ["biomarkers", "pathology"]:
-        return transform.immunohisto_transform(
-            datasets=datasets, dataset_name=dataset_name
-        )
-    if dataset_name == "model_details":
-        return transform.transform_model_details(datasets=datasets)
+    if isinstance(function_info, str):
+        function_name = function_info
+        config_defined_params = {}
     else:
-        return None
+        # Retrieve the function name and its parameters
+        # Assumes a single function in the dict
+        function_name, config_defined_params = next(iter(function_info.items()))
+    try:
+        retrieved_function = getattr(transform, function_name)
+    except AttributeError:
+        raise AttributeError(
+            f"Function {function_name} not found in the transform module. Please provide the correct function name."
+        )
+    sig = inspect.signature(retrieved_function)
+    parameters = {}
+    # Map known inputs
+    for name, _ in sig.parameters.items():
+        if name == "df":
+            df = datasets[dataset_name]
+            parameters["df"] = df
+        elif name == "datasets":
+            parameters["datasets"] = datasets
+        elif name == "dataset_name":
+            parameters["dataset_name"] = dataset_name
+    # Holds all the parameters to be passed to the transformation function
+    combined_params = {**parameters, **config_defined_params}
+    return retrieved_function(**combined_params)
```

Comment:
I think this approach is fine as written, but because we were going for a more mapping-based approach without so much nesting/if/else's, I wonder if we should do something where we set up a map like
```
standard_params = {
    "df": datasets[dataset_name],
    "datasets": datasets,
    "dataset_name": dataset_name,
}
```
which we can refer to when we loop through the params.

I also think you can simplify that loop somewhat by grabbing the function parameters specifically right off the bat like
```
function_params = inspect.signature(retrieved_function).parameters
```

Reply:
Plus one on this suggestion